### PR TITLE
fix(rehype-typst): inline math not rendering

### LINKED ISSUE #603: rehype-typst broken inline-math typst code
**Describe the bug**
rehype-typst fails to render inline-math 

**To Reproduce**
Minimal steps to reproduce the behavior (Library test):
1. create file `example.js`
```js
import rehypeTypst from '@myriaddreamin/rehype-typst'
import rehypeParse from 'rehype-parse'
import rehypeStringify from 'rehype-stringify'
import {read, write} from 'to-vfile'
import {unified} from 'unified'

const file = await unified()
  .use(rehypeParse, {fragment: true})
  .use(rehypeTypst)
  .use(rehypeStringify)
  .process(await read('input.html'))

file.basename = 'output.html'
await write(file)
```
2. create file `input.html`
 ```html
 <code class="language-math">x</code>
```
3. install packages (see documentation)
4. run with `node example.js`, see error resulting from typst failing to compile the `inlineMathTemplate`.

**Package/Software version (using command `typst-ts-cli --VV full` and paste down the output):**
```plain
rehype-typst "0.5.0-rc7
typst.ts 0.5.0-rc9
rehype-parse 9.0.1
rehype-stringify 10.0.1
to-vfile 8.0.0
unified 11.0.5
```

**Additional context**
[Typst 0.12](https://typst.app/blog/2024/typst-0.12/) deprecated various functions such as `locate` in favour of `context`. This can be fixed by editing `inlineMathTemplate`. See coming PR

### PR DESCRIPTION
**Pull request by kimhanm**
fix #603

### CODE REVIEW

**Comment by kimhanm** [comment]
See #603

File: projects/rehype-typst/lib/index.js [hunk 1]
```diff
@@ -113,7 +113,7 @@ export default function rehypeTypst(options) {
         const defaultEm = 11;
         const height = parseFloat(root.children[0].properties['dataHeight']);
         const width = parseFloat(root.children[0].properties['dataWidth']);
-        const shift = height - result.baselinePosition;
+        const shift = height - result.baselinePosition + 2.8; /* line up baselines */
```
Comment (aarnphm) [inline projects/rehype-typst/lib/index.js:116]: this seems like a hack fwiw. is this documented anywhere?

Comment (aarnphm) [reply projects/rehype-typst/lib/index.js:116]: yeah this seems hacky, given that it depends on users' `line-height`, so this doesn't makes sense to me.

probably a better solution is to target the CSS class for the result rendered equation.

**Comment by Myriad-Dreamin** [comment]
@Enter-tainer

**Comment by Enter-tainer** [comment]
Hi @kimhanm! Thank you for updating the template! I just did a quick fix based on your commit and it seems that it's working now. image

**Review by Enter-tainer — APPROVED** [review]

**Pull request merged**